raw_data_QA/check_raw_case_death_data.py

```diff
@@ -0,0 +1,658 @@
+        "--new_deaths_name",
+        type=str,
+        dest="new_deaths_name",
+        default="deaths",
+        help="name of new deaths name in input csv",
+    )
+    parser.add_argument(
+        "-date_name",
+        "--date_name",
+        type=str,
+        dest="date_name",
+        default="date",
+        help="name of datesin input csv",
+    )
+    parser.add_argument(
+        "-updated_date_name",
+        "--updated_date_name",
+        type=str,
+        dest="updated_date_name",
+        default="Date",
+        help="name of date variable that is computed in this script, really just renaming",
+    )
+    parser.add_argument(
+        "-rmse_threshold",
+        "--rmse_threshold",
+        type=float,
+        dest="rmse_threshold",
+        default=1,
+        help="rmse threshold to trigger on",
+    )
+    parser.add_argument(
+        "-percent_threshold",
+        "--percent_threshold",
+        type=float,
+        dest="percent_threshold",
+        default=5,
+        help="percent difference threshold to trigger on",
+    )
+    parser.add_argument(
+        "-z_threshold",
+        "--z_threshold",
+        type=float,
+        dest="z_threshold",
+        default=2,
+        help="Z score threshold to trigger on",
+    )
+    parser.add_argument(
+        "-n_days_over_threshold",
+        "--n_days_over_threshold",
+        dest="n_days_over_threshold",
+        type=int,
+        default=5,
+        help="number of days to be over threshold to trigger on",
+    )
+    parser.add_argument(
+        "-use_latest",
+        "--use_latest",
+        type=bool,
+        default=False,
+        help="set to true to use latest data * and * new data to determine unexpected changes",
+    )
+    parser.add_argument(
+        "-n_days_z_score_mean",
+        "--n_days_z_score_mean",
+        type=int,
+        default=14,
+        help="how many days in the past to average the z score to look for unexpcted changes to the data",
+    )
+    parser.add_argument(
+        "-new_day_percent_thres",
+        "--new_day_percent_thres",
+        type=float,
+        dest="new_day_percent_thres",
+        default=10,
+        help="percent change threshold for new days to trigger on",
+    )
+    parser.add_argument(
+        "-covid_data_public_dir",
+        "--covid_data_public_dir",
+        type=str,
+        dest="covid_data_public_dir",
+        default="../../covid-data-public",
+        help="directory of covid-data-public",
+    )
+    parser.add_argument(
+        "-data_source",
+        "--data_source",
+        type=str,
+        dest="data_source",
+        default="NYT",
+        help="input data source (e.g. NYT/JHU)",
+    )
+    args = parser.parse_args()
+    # put output dir in current working dir
+    # args.output_dir = working_dir() + "/" + args.output_dir
+    # Create separate output folders based on abnormality of data
+    args.new_data_abnormal_folder = "new_data_abnormal"
```

Comment:
It is handy have an object to pass around that holds all the arguments needed in various bits of code, but args is unstructured. A dataclass makes it easier for future readers of the code to know what is available and in this project we're using pydantic to create them. Let me know if you'd like more help conforming.

Reply:
yeah I think rn we just need something ready, so refactoring the code would be for v2